eclipse-collections/src/main/java/org/eclipse/collections/impl/list/immutable/ImmutableArrayList.java

```diff
@@ -603,10 +603,11 @@ public ImmutableList<T> take(int count)
     @Override
     public ImmutableList<T> takeWhile(Predicate<? super T> predicate)
     {
-        int endIndex = this.detectNotIndex(predicate);
-        T[] result = (T[]) new Object[endIndex];
-        System.arraycopy(this.items, 0, result, 0, endIndex);
-        return new ImmutableArrayList<>(result);
+        int startIndex = this.detectNotIndex(predicate);
+        int resultSize = this.size() - startIndex;
+        T[] result = (T[]) new Object[resultSize];
+        System.arraycopy(this.items, startIndex, result, 0, resultSize);
+        return Lists.immutable.with(result);
```

Comment:
This is an implementation of `dropWhile`, it needs to be restored to `takeWhile`